### ISSUE TITLE
Support Content-Security-Policy without "script-src unsafe-inline"

### ISSUE THREAD

**Issue by graingert** [divio/django-cms#5381]
Django 1.10 lands support for running the admin with a Content-Security-Policy without "script-src unsafe-inline" applied.

Django CMS should follow suit and remove all inline JavaScript. Replacing large blocks of JS with `<script src={% static 'path/to/old_inlines/eg.js' %}></script>` and `<script>var foo = {{bar}}></script>` with:
```js
# path/to/fooUsage.js
window.foo = $('#django-cms-foo').data('foo');
```
```html
<script id='django-cms-foo' src={% static 'path/to/fooUsuage' %} data-foo={{ bar }}></script>
```

**Comment by vxsx** [comment]
Thanks for the report, we'll take care of it. 
It would also require changes in some of the essential addons though, e.g. djangocms-text-ckeditor etc.

**Comment by yakky** [comment]
I think that this should undergo a deprecation period. Django 1.10 just removed its own inline javascript to allow enabling "Content-Security-Policy", it does not enforce it.
I think that django CMS 3.4 can adopt the same behavior, and then let addons adapt to it by providing a "suggested timeframe"

**Comment by rspeed** [comment]
I gave this a bit of a look and it's looking a bit harder than I expected. Replacing `CMS.config` with a block of JSON was fairly straightforward, but plugins are another story altogether. Gonna put it aside for now and take another look when I have some time.

**Comment by graingert** [comment]
That's very unfortunate

**Comment by rspeed** [comment]
Some good news: I got the plugins/placeholders sorted out. I've got it running on my personal site right now (with `style-src 'self' 'unsafe-inline'` in the CSP) and haven't encountered any issues.

**Comment by graingert** [comment]
@rspeed oh great stuff, any chance of some PRs?

**Comment by rspeed** [comment]
Yup, already working on one. I found a few more inline scripts that needed to be fixed. Finishing that up now.

**Comment by marksweb** [comment]
@vxsx @yakky what's the status of CSP support?

Following a PEN test I've got a client who wants us to implement a CSP & I've done what I can, but can't complete the task without support from the CMS.

I've not looked at the commits involved in #6072 yet, but it's full of conflicts given it's age now so not sure how viable those changes are now.
